src/components/settings/SettingsLMSTab/ErrorReporting/ContentMetadataTable.jsx

```diff
@@ -9,38 +11,81 @@ import DownloadCsvButton from './DownloadCsvButton';
 import { getSyncStatus, getSyncTime } from './utils';
 
 function ContentMetadataTable({ config, enterpriseCustomerUuid }) {
-  const [data, setData] = useState([]);
+  const [currentPage, setCurrentPage] = useState();
+  const [currentFilters, setCurrentFilters] = useState();
+  const [paginationData, setPaginationData] = useState({
+    itemCount: 0,
+    pageCount: 0,
+    data: [],
+    sortBy: '',
+  });
 
   useEffect(() => {
     const fetchData = async () => {
       const response = await LmsApiService.fetchContentMetadataItemTransmission(
-        enterpriseCustomerUuid, config.channelCode, config.id,
+        enterpriseCustomerUuid, config.channelCode, config.id, currentPage, currentFilters,
       );
       return response;
     };
 
     fetchData()
       .then((response) => {
-        setData(response.data.results);
+        setPaginationData({
+          itemCount: response.data.count,
+          pageCount: response.data.pages_count,
+          data: response.data.results,
+        });
       })
       .catch((err) => {
         logError(err);
       });
-  }, [config.channelCode, config.id, enterpriseCustomerUuid]);
+  }, [config.channelCode, config.id, enterpriseCustomerUuid, currentPage, currentFilters]);
+
+  // Call back function, handles filters and page changes
+  const fetchData = useCallback(
+    (args) => {
+      const newFiltersMap = args.filters.map((filter) => ({ [filter.id]: filter.value }));
+      const newFilters = Object.assign({}, ...newFiltersMap);
+
+      const newSortByMap = args.sortBy.map((filter) => ({ sort_by: `${filter.desc ? '-' : ''}${filter.id}` }));
+      const newSortBys = Object.assign(newFilters, ...newSortByMap);
+
+      setCurrentFilters({ ...newFilters, ...newSortBys });
+
+      // pages index from 1 in backend, DataTable component index from 0
+      if (args.pageIndex !== currentPage - 1) {
+        setCurrentPage(args.pageIndex + 1);
+      }
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [setCurrentPage],
+  );
+
+  const memoisedFetchData = useMemo(() => ((args) => fetchData(args)), [fetchData]);
 
   return (
     <div className="pt-4">
       <DataTable
         isSortable
         isFilterable
         numBreakoutFilters={2}
+        manualFilters
         defaultColumnValues={{ Filter: TextFilter }}
         isPaginated
-        itemCount={data?.length}
-        data={data}
+        manualPagination
+        pageCount={paginationData.pageCount}
+        itemCount={paginationData.itemCount}
+        fetchData={memoisedFetchData}
+        manualSortBy
+        initialState={{
+          pageSize: 10,
+          pageIndex: currentPage ? currentPage - 1 : 0,
```

Comment:
Since the front-end component is using 0-based indexes, and the backend api uses 1-based indexes, maybe it makes sense to convert the indexes during the send/receive steps so we can keep consistency in the js code?

Reply:
Perhaps, a lot of the pagination is automatically handled by Django on the backend. Maybe I can tweak it in the api client here on the frontend.

Reply:
done! Put it in the api client